lab_3_generate_by_ngrams/main.py

```diff
@@ -129,6 +187,13 @@ def fill_from_ngrams(self, content: dict) -> None:
         Args:
             content (dict): ngrams from external JSON
         """
+        if not isinstance(content, dict) or not content:
+            return None
+        for element in content['freq']:
+            for token in element:
+                if token.isalpha():
```

Comment:
Notice that you miss the spaces which should be converted to EoW token

Otherwise good, that's a method for 10!

Reply:
Done